driver/src/price_finding/linear_optimization_price_finder.rs

```diff
@@ -94,10 +93,10 @@ fn deserialize_result(
                 .unwrap_or_else(|| (token.to_owned(), "0".to_string()))
         })
         .collect::<Prices>();
-    let prices = (0..num_tokens)
+    let prices = (0..price_map.len())
```

Comment:
I'm not a huge fan of the unwrap(). Can we comment why this is not going to panic (price_map can have at most uint16 prices) or maybe even allow tokenId() to be called with usize as argument?

Reply:
I would be in favour of allowing `token_id()` be called on `usize`